--- Mbstring.php	2021-03-13 21:53:56.167484655 +0100
+++ Mbstring.php	2021-03-13 22:32:12.786189176 +0100
@@ -65,11 +65,12 @@
  *
  * @internal
  */
 final class Mbstring
 {
-    public const MB_CASE_FOLD = \PHP_INT_MAX;
+    // "public const" requires PHP 7.1
+    /*public*/ const MB_CASE_FOLD = \PHP_INT_MAX;
 
     private static $encodingList = ['ASCII', 'UTF-8'];
     private static $language = 'neutral';
     private static $internalEncoding = 'UTF-8';
     private static $caseFold = [
@@ -733,10 +734,11 @@
 
         if ('UTF-8' !== $encoding) {
             $s = iconv($encoding, 'UTF-8//IGNORE', $s);
         }
 
+	$wide = -1;
         $s = preg_replace('/[\x{1100}-\x{115F}\x{2329}\x{232A}\x{2E80}-\x{303E}\x{3040}-\x{A4CF}\x{AC00}-\x{D7A3}\x{F900}-\x{FAFF}\x{FE10}-\x{FE19}\x{FE30}-\x{FE6F}\x{FF00}-\x{FF60}\x{FFE0}-\x{FFE6}\x{20000}-\x{2FFFD}\x{30000}-\x{3FFFD}]/u', '', $s, -1, $wide);
 
         return ($wide << 1) + iconv_strlen($s, 'UTF-8');
     }
 
